duckduckgo/macos-browser
# Improve Sync-related database cleaning logic

Task/Issue URL: https://app.asana.com/0/414235014887631/1205315938732113/f
Tech Design URL:
CC:

**Description**:
Only clean database when sync is inactive at app startup.

**Steps to test this PR**:
1. Run the app from Xcode while Sync is not active
2. Verify that bookmarks and credentials database cleanup was attempted ("No bookmarks pending deletion" and "No syncable credentials metadata pending deletion" messages appear in the log).
3. Enable Sync, verify that bookmarks and credentials cleanup was attempted, then quit the app.
4. Relaunch the app, verify that bookmarks and credentials cleanup was **not** attempted.
5. Disable Sync, verify that bookmarks and credentials cleanup was attempted.



---
###### Internal references:
[Pull Request Review Checklist](https://app.asana.com/0/1202500774821704/1203764234894239/f)
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
[Pull Request Documentation](https://app.asana.com/0/1202500774821704/1204012835277482/f)
